wavedump/src/wavedump.c

```diff
@@ -1383,8 +1407,41 @@ int add_to_output_file(char *filename, float data[WF_SIZE][32][1024], int n, int
         H5Sclose(aid);
         H5Aclose(attr);
         H5Tclose(atype);
+        
+	/* Writing the baseline data to the file
+	 * in separate datasets */
+	for (i = 0; i < 32; i++) {
+            if (!(chmask & (1 << i))) continue;
+      	    maxdims[0] = BS_SIZE;
+            maxdims[1] = nsamples;
+            dims[0] = BS_SIZE;
+            dims[1] = nsamples;
+            
+	    space = H5Screate_simple(2, dims, maxdims);
+            
+            float wdata[BS_SIZE][1024];
+	    for (j = 0; j < BS_SIZE; j++)
+                for (k = 0; k < nsamples; k++)
+                    wdata[j][k] = baseline_data[j][i][k];
+
+            sprintf(dset_name, "base_ch%i", i);
+            
+	    dset = H5Dcreate(file, dset_name, H5T_NATIVE_FLOAT, space, H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
 
-        for (i = 0; i < 32; i++) {
+            /* Write the data to the dataset. */
+            status = H5Dwrite(dset, H5T_NATIVE_FLOAT, H5S_ALL, H5S_ALL, H5P_DEFAULT, wdata);
+
+            /* Close and release resources. */
+            status = H5Dclose(dset);
+            status = H5Sclose(space);
+
+            if (status) {
+                fprintf(stderr, "error closing hdf5 resources.\n");
+                return 1;
+            }
+        }

```

Comment:
seem to be some weird indentations here. In vim you can do:

```
:set shiftwidth=4
:set softtabstop=4
:set expandtab
```

and then I think you can block highlight the whole section and run `:retab`, then you should be able to see the indendation problems.